peachjam/adapters/adapters.py

```diff
@@ -290,6 +292,21 @@ def update_document(self, url):
             # the source file is the PDF version
             self.download_source_file(f"{url}.pdf", created_doc, title)
 
+        if document["taxonomy_topics"]:
+            # get topics beginning with "subject-areas"
+            topics = [
+                t for t in document["taxonomy_topics"] if t.startswith("subject-areas")
+            ]
+            if topics:
+                taxonomies = Taxonomy.objects.filter(slug__in=topics)
+                created_doc.taxonomies.all().delete()
```

Comment:
Check the proposal -- you need to delete only those taxonomies that are part of `subject-areas` -- otherwise manually added taxonomies will be lost.

Basically, this adapter only handles taxonomies under a certain topic, and should only play within those walls.

Reply:
Ah, right. Updated.